pkg/services/annotations/annotationsimpl/annotations.go

```diff
@@ -72,12 +72,45 @@ func (r *RepositoryImpl) Update(ctx context.Context, item *annotations.Item) err
 }
 
 func (r *RepositoryImpl) Find(ctx context.Context, query *annotations.ItemQuery) ([]*annotations.ItemDTO, error) {
-	resources, err := r.authZ.Authorize(ctx, query.OrgID, query)
-	if err != nil {
-		return make([]*annotations.ItemDTO, 0), err
+	if query.Limit == 0 {
+		query.Limit = 1000
 	}
 
-	return r.reader.Get(ctx, query, resources)
+	// Return early if no annotations found, it's not necessary to perform expensive access control filtering
+	res, err := r.reader.Get(ctx, query, &accesscontrol.AccessResources{
+		Dashboards:               map[string]int64{},
```

Comment:
I would think it's ok to not pass this (`nil` maps still have a len of 0), unless you want it to more obviously show that this is not filtering by dashboard.

Reply:
The problem is that underlying function (`getAccessControlFilter()`) does not check for nil and then access `Dashboards` field, so this might cause a panic. I think I can also add nil check there).